src/Uno.UI/UI/Xaml/Controls/DatePicker/DatePicker.cs

```diff
@@ -294,46 +361,110 @@ private void BindToFlyout(string propertyName)
 			this.Binding(propertyName, propertyName, _flyoutButton.Flyout, BindingMode.TwoWay);
 		}
 
-		private void InitializeTextBlocks(IFrameworkElement container)
+		private void ReassignColumns()
 		{
-			if (container.GetTemplateChild(FlyoutButtonGridPartName) is Grid grid)
+			if (_flyoutGrid == null)
+			{
+				return;
+			}
+			/* DatePicker normally contains 3 textblocks meant for day/month/year with a different Grid.Column each.
+			 * We need to shuffle the columns with the order dictated by current culture, eg: yyyy/mm/dd vs mm/dd/yyyy...
+			 * Since we can't just "move" the column, we have to move the textblocks' column, and
+			 * swap the relevant properties (eg: ColumnDefinition.Width) from the old column to the new one. */
+			var orderedColumns = GetOrderedTextBlocksForCulture(CultureInfo.CurrentCulture);
```

Comment:
these changes need some tests - you can probably add some `Uno.UI.RuntimeTest` with different cultures+visibilities, and programmatically check that the column values are as expected